automap/src/mocks.rs

```diff
@@ -200,3 +219,227 @@ impl FindRoutersCommandMock {
+            delete_mapping_results: RefCell::new(vec![]),
+            start_housekeeping_thread_params: Arc::new(Mutex::new(vec![])),
+            start_housekeeping_thread_results: RefCell::new(vec![]),
+            stop_housekeeping_thread_params: Arc::new(Mutex::new(vec![])),
+            stop_housekeeping_thread_results: RefCell::new(vec![]),
+        }
+    }
+
+    pub fn find_routers_result(self, result: Result<Vec<IpAddr>, AutomapError>) -> Self {
+        self.find_routers_results.borrow_mut().push(result);
+        self
+    }
+
+    pub fn get_public_ip_params(mut self, params: &Arc<Mutex<Vec<IpAddr>>>) -> Self {
+        self.get_public_ip_params = params.clone();
+        self
+    }
+
+    pub fn get_public_ip_result(self, result: Result<IpAddr, AutomapError>) -> Self {
+        self.get_public_ip_results.borrow_mut().push(result);
+        self
+    }
+
+    #[allow(clippy::type_complexity)]
+    pub fn add_mapping_params(mut self, params: &Arc<Mutex<Vec<(IpAddr, u16, u32)>>>) -> Self {
+        self.add_mapping_params = params.clone();
+        self
+    }
+
+    pub fn add_mapping_result(self, result: Result<u32, AutomapError>) -> Self {
+        self.add_mapping_results.borrow_mut().push(result);
+        self
+    }
+
+    pub fn add_permanent_mapping_params(mut self, params: &Arc<Mutex<Vec<(IpAddr, u16)>>>) -> Self {
+        self.add_permanent_mapping_params = params.clone();
+        self
+    }
+
+    pub fn add_permanent_mapping_result(self, result: Result<u32, AutomapError>) -> Self {
+        self.add_permanent_mapping_results.borrow_mut().push(result);
+        self
+    }
+
+    pub fn delete_mapping_params(mut self, params: &Arc<Mutex<Vec<(IpAddr, u16)>>>) -> Self {
+        self.delete_mapping_params = params.clone();
+        self
+    }
+
+    pub fn delete_mapping_result(self, result: Result<(), AutomapError>) -> Self {
+        self.delete_mapping_results.borrow_mut().push(result);
+        self
+    }
+
+    pub fn start_housekeeping_thread_result(
+        self,
+        result: Result<Sender<HousekeepingThreadCommand>, AutomapError>,
+    ) -> Self {
+        self.start_housekeeping_thread_results
+            .borrow_mut()
+            .push(result);
+        self
+    }
+
+    pub fn start_housekeeping_thread_params(
+        mut self,
+        params: &Arc<Mutex<Vec<(ChangeHandler, IpAddr)>>>,
+    ) -> Self {
+        self.start_housekeeping_thread_params = params.clone();
+        self
+    }
+
+    pub fn stop_housekeeping_thread_params(mut self, params: &Arc<Mutex<Vec<()>>>) -> Self {
+        self.stop_housekeeping_thread_params = params.clone();
+        self
+    }
+
+    pub fn stop_housekeeping_thread_result(
+        self,
+        result: Result<ChangeHandler, AutomapError>,
+    ) -> Self {
+        self.stop_housekeeping_thread_results
+            .borrow_mut()
+            .push(result);
+        self
+    }
+}
+
+pub fn pmp_protocol_scenario_for_actor_system_factory_test(
```

Comment:
Maybe this could be something like "prepare_test_automap_control" that accepted a bunch of parameters including mock Transactors and a change handler. Then you could prepare the mock Transactors in the distant test, and use this just for final assembly of the AutomapControlReal, without this function having specific knowledge of what the distant test requires.